src/libsyntax/attr.rs

```diff
@@ -77,6 +81,93 @@ pub fn is_used(attr: &Attribute) -> bool {
     })
 }
 
+pub trait AttrMetaListItemMethods {
```

Comment:
The issues is that AttrMetaMethod's `name()` always returns an InternedString, while ListMetaMethods returns an Option<...>.